x-pack/legacy/plugins/apm/public/context/MatchedRouteContext.tsx

```diff
@@ -21,11 +24,18 @@ export function MatchedRouteProvider({
   const { pathname } = useLocation();
 
   const contextValue = useMemo(() => {
-    return routes.filter(route => {
-      return matchPath(pathname, {
-        path: route.path
+    let matchedPath: Match | undefined;
+    const [matchedRoute] = routes.filter(route => {
+      const path = matchPath(pathname, {
+        path: route.path,
+        exact: true
       });
+      if (path) {
+        matchedPath = path;
+      }
+      return path;
     });
+    return { matchedRoute, matchedPath };
```

Comment:
I think `matchedPath` is much better than `matchPath`, it makes sense now. Still, considering that we now call the breadcrumb function in multiple places, what do you think of either re-using what we have today, or, even better IMHO, refactoring this to return the breadcrumb on the route object (basically `{...route, breadcrumb }` and then using this hook in `ProvideBreadcrumbs.tsx`?